metadata_mapper/mappers/validator.py

```diff
@@ -218,15 +218,27 @@ def set_validatable_fields(self, fields: list[dict] = [],
         If merge is False, simply sets the fields to the provided list.
 
         Parameters:
-            fields: dict
-                A dict containing validation definitions.
+            fields: list[dict[str, Any]] (default: [])
+                A list of dicts containing validation definitions.
+                See default_validatable_fields for examples.
             merge: bool (default: True)
                 Should `fields` be merged into `default_validatable_fields`?
 
         Returns list[dict]
         """
         if merge:
-            self.validatable_fields = [*default_validatable_fields, *fields]
+            self.validatable_fields = default_validatable_fields
+
+            for field in fields:
+                index = [
```

Comment:
what is happening here?

Reply:
I read this as "index is an array of tuples, (i, el), for i el in enumerate default_validatable_fields if field['field'] == el['field']"

If index is an array of tuples, how do you use it as an index on line 239?

Reply:
I agree this was too convoluted. I've simplified it a bit. Now, it performs a simple dict comprehension so it can merge (ensuring unique field names), then converts back to a list.

Refactoring validatable_fields to be a dict is an option, too, but keeping it as a list of dicts makes iterating over it a bit simpler.